src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -49,35 +45,15 @@ const NavHeader = (props) => {
     onCancelToken,
     placeholderText,
     setIsFirstTimeOpen,
-    getSearchLayout,
-    opacityImage
+    opacityImage,
+    onFollowButtonPress,
+    followType
   } = props;
   const navigation = useNavigation();
   const coverPath = topicDetail?.cover_path || null;
   const insets = useSafeAreaInsets();
   const {width: displayWidth} = useWindowDimensions();
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleFollowTopic` function has been removed and replaced with a prop function `onFollowButtonPress`. This change simplifies the component by moving the logic for following a topic outside of it. However, ensure that the parent component passes the correct function as a prop.
```diff
-  const {followTopic} = useChatClientHook();
-
-  const handleFollowTopic = async () => {
-    setIsFollow(!isFollow);
-    if (isFollow) {
-      setMemberCount(memberCount - 1);
-    } else {
-      setMemberCount(memberCount + 1);
-    }
-    try {
-      const followed = await followTopic(topicDetail?.name);
-
-      setIsFollow(followed);
-      getTopicDetail(topicDetail?.name);
-    } catch (error) {
-      if (__DEV__) {
-        console.log(error);
-      }
-    }
-  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->